src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -455,9 +455,12 @@ protected void SelectVersion()
                 (_versions.Any(v => v != null && !v.IsValidVersion) &&
                     _versions.IndexOf(SelectedVersion) > _versions.IndexOf(_versions.FirstOrDefault(v => v != null && !v.IsValidVersion))))
             {
-                // it should always select the top version from versions list to install or update
-                // which has a valid version. If find none, then just set to null.
-                SelectedVersion = _versions.FirstOrDefault(v => v != null && v.IsValidVersion);
+                // Select the installed version by default.
+                // Otherwise, select the first version in the version list.
+                var possibleVersions = _versions.Where(v => v != null);
+                SelectedVersion =
+                    possibleVersions.FirstOrDefault(v => v.Version.Equals(_searchResultPackage.InstalledVersion))
```

Comment:
In /NuGet/Home#8679 and PR #3084, we modified code to use `NuGetVersion` comparer instead of using `System.Version` comparer. If I am not wrong it looks like this line of code is still pointing to `System.Version` comparer

Reply:
Here, `_versions` is a `List<DisplayVersion>`, and `DisplayVersion.Version` is a `NuGetVersion`. In the other PR the collection was of `NuGetVersion`,